### ISSUE TITLE
np.float64 in binstats output causes ast error (fix included)

### ISSUE THREAD

**Issue by george-coulouris** [Ecogenomics/CheckM#396]
bin_stats.analyze.tsv contains np.float64:
> annotation      {'GC': 0.5217391304347826, 'GC std': 0.0, 'Genome size': 87, '# ambiguous bases': 3, '# scaffolds': 1, '# contigs': 1, 'Longest scaffold': 87, 'Longest contig': 84, 'N50 (scaffolds)': 87, 'N50 (contigs)': 84, 'Mean scaffold length': np.float64(87.0), 'Mean contig length': np.float64(84.0), 'Coding density': -1, 'Translation table': -1, '# predicted genes': -1}

which ast cannot parse in parseBinStats:
```
Unexpected error: <class 'ValueError'>
...
  File "/home/coulouri/checkm_venv/lib/python3.9/site-packages/checkm/resultsParser.py", line 157, in parseBinStats
    binStats[lineSplit[0]] = ast.literal_eval(lineSplit[1])
...
  File "/opt/python-3.9/lib/python3.9/ast.py", line 66, in _raise_malformed_node
    raise ValueError(f'malformed node or string: {node!r}')
```

the following modification to binStatistics.py causes native float types to be emitted rather than numpy float64, which then allows ast to parse successfully:
```
            binStats['Mean scaffold length'] = scaffoldAvgLen.self()
            binStats['Mean contig length'] = contigAvgLen.self()
```

Thanks for fixing!

**Comment by matinnuhamunada** [comment]
I tried the solution above but I cannot call `self()` from `scaffoldAvgLen` and `contigAvgLen`?

```bash
Traceback (most recent call last):
  File "/home/matinnu/test/bgcflow/.snakemake/conda/9155ccccbb351d3788fd4776665701e0_/lib/python3.11/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/home/matinnu/test/bgcflow/.snakemake/conda/9155ccccbb351d3788fd4776665701e0_/lib/python3.11/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/home/matinnu/test/bgcflow/.snakemake/conda/9155ccccbb351d3788fd4776665701e0_/lib/python3.11/site-packages/checkm/binStatistics.py", line 130, in _processBin
    binStats['Mean scaffold length'] = scaffoldAvgLen.self()
                                       ^^^^^^^^^^^^^^^^^^^
AttributeError: 'numpy.float64' object has no attribute 'self'
```

Changing numpy float64 to simple float in `binStatistics.py` also works:
```python
            binStats['Mean scaffold length'] = float(scaffoldAvgLen)
            binStats['Mean contig length'] = float(contigAvgLen)
```
<details>
<summary>here's the conda environment:</summary>

```yaml
name: checkm
channels:
  - bioconda
  - conda-forge
dependencies:
  - _libgcc_mutex=0.1=conda_forge
  - _openmp_mutex=4.5=2_gnu
  - brotli=1.1.0=hd590300_1
  - brotli-bin=1.1.0=hd590300_1
  - bzip2=1.0.8=hd590300_5
  - c-ares=1.28.1=hd590300_0
  - ca-certificates=2024.6.2=hbcca054_0
  - certifi=2024.6.2=pyhd8ed1ab_0
  - checkm-genome=1.2.2=pyhdfd78af_1
  - contourpy=1.2.1=py311h9547e67_0
  - cycler=0.12.1=pyhd8ed1ab_0
  - dendropy=5.0.1=pyhdfd78af_0
  - fonttools=4.53.0=py311h331c9d8_0
  - freetype=2.12.1=h267a509_2
  - gettext=0.22.5=h59595ed_2
  - gettext-tools=0.22.5=h59595ed_2
  - hmmer=3.4=hdbdd923_1
  - keyutils=1.6.1=h166bdaf_0
  - kiwisolver=1.4.5=py311h9547e67_1
  - krb5=1.21.2=h659d440_0
  - lcms2=2.16=hb7c19ff_0
  - ld_impl_linux-64=2.40=hf3520f5_7
  - lerc=4.0.0=h27087fc_0
  - libasprintf=0.22.5=h661eb56_2
  - libasprintf-devel=0.22.5=h661eb56_2
  - libblas=3.9.0=22_linux64_openblas
  - libbrotlicommon=1.1.0=hd590300_1
  - libbrotlidec=1.1.0=hd590300_1
  - libbrotlienc=1.1.0=hd590300_1
  - libcblas=3.9.0=22_linux64_openblas
  - libcurl=8.8.0=hca28451_0
  - libdeflate=1.20=hd590300_0
  - libedit=3.1.20191231=he28a2e2_2
  - libev=4.33=hd590300_2
  - libffi=3.4.2=h7f98852_5
  - libgcc-ng=13.2.0=h77fa898_11
  - libgettextpo=0.22.5=h59595ed_2
  - libgettextpo-devel=0.22.5=h59595ed_2
  - libgfortran-ng=13.2.0=h69a702a_11
  - libgfortran5=13.2.0=h3d2ce59_11
  - libgomp=13.2.0=h77fa898_11
  - libiconv=1.17=hd590300_2
  - libidn2=2.3.7=hd590300_0
  - libjpeg-turbo=3.0.0=hd590300_1
  - liblapack=3.9.0=22_linux64_openblas
  - libnghttp2=1.58.0=h47da74e_1
  - libnsl=2.0.1=hd590300_0
  - libopenblas=0.3.27=pthreads_h413a1c8_0
  - libpng=1.6.43=h2797004_0
  - libsqlite=3.46.0=hde9e2c9_0
  - libssh2=1.11.0=h0841786_0
  - libstdcxx-ng=13.2.0=hc0a3c3a_11
  - libtiff=4.6.0=h1dd3fc0_3
  - libunistring=0.9.10=h7f98852_0
  - libuuid=2.38.1=h0b41bf4_0
  - libwebp-base=1.4.0=hd590300_0
  - libxcb=1.16=hd590300_0
  - libzlib=1.3.1=h4ab18f5_1
  - matplotlib-base=3.8.4=py311ha4ca890_2
  - munkres=1.1.4=pyh9f0ad1d_0
  - ncurses=6.5=h59595ed_0
  - numpy=2.0.0=py311h1461c94_0
  - openjpeg=2.5.2=h488ebb8_0
  - openssl=3.3.1=h4ab18f5_0
  - packaging=24.1=pyhd8ed1ab_0
  - pillow=10.3.0=py311h82a398c_1
  - pip=24.0=pyhd8ed1ab_0
  - pplacer=1.1.alpha19=h9ee0642_2
  - prodigal=2.6.3=h031d066_9
  - pthread-stubs=0.4=h36c2ea0_1001
  - pyparsing=3.1.2=pyhd8ed1ab_0
  - pysam=0.22.1=py311h1f0e11c_1
  - python=3.11.0=he550d4f_1_cpython
  - python-dateutil=2.9.0=pyhd8ed1ab_0
  - python_abi=3.11=4_cp311
  - readline=8.2=h8228510_1
  - scipy=1.13.1=py311h517d4fd_0
  - setuptools=70.1.0=pyhd8ed1ab_0
  - six=1.16.0=pyh6c4a22f_0
  - tar=1.34=hb2e2bae_1
  - tk=8.6.13=noxft_h4845f30_101
  - tzdata=2024a=h0c530f3_0
  - wget=1.21.4=hda4d442_0
  - wheel=0.43.0=pyhd8ed1ab_1
  - xorg-libxau=1.0.11=hd590300_0
  - xorg-libxdmcp=1.1.3=h7f98852_0
  - xz=5.2.6=h166bdaf_0
  - zlib=1.3.1=h4ab18f5_1
  - zstd=1.5.6=ha6fb4c9_0
```

</details>

**Comment by matinnuhamunada** [comment]
Pinning `numpy` to version `1.26.4` resolves the issue. 

This behaviour change seems to be the result of the recent [`numpy` release (`v2.0.0`)](https://numpy.org/doc/stable/release/2.0.0-notes.html).

**Comment by donovan-h-parks** [comment]
Hi. Thanks for digging into this. I am making this change now and will push an updated version of CheckM later today.
